nexus/db-queries/src/db/datastore/switch_interface.rs

```diff
@@ -130,7 +133,7 @@ impl DataStore {
         self.transaction_retry_wrapper("loopback_address_delete")
             .transaction(&conn, |conn| async move {
                 let la = diesel::delete(dsl::loopback_address)
-                    .filter(dsl::id.eq(id))
+                    .filter(dsl::id.eq(to_db_typed_uuid(id)))
```

Comment:
I looked into "what would it take to not need to specify this converter wrapper", and frankly, I think you've made the right call. 

[`.eq`](https://docs.diesel.rs/master/diesel/expression_methods/trait.ExpressionMethods.html#method.eq) requires the argument type to have `T: AsExpression<Self::SqlType>`, which can either be derived (as you've done for the `DbTypedUuid`, or which is implemented generically:

```rust
impl<T, ST> AsExpression<ST> for T
where
  T: Expression<SqlType = ST>,
  ST: SqlType + TypedExpressionType`
```

([source](https://docs.diesel.rs/master/diesel/expression/trait.AsExpression.html#impl-AsExpression%3CST%3E-for-T))

I think this is how "Uuid" works natively here -- it implements `SqlType + TypedExpressionType`.
If we wanted to, we probably *could* implement either `AsExpression` or `SqlType` for `TypedUuid<T>`, but then it forces the UUID registry to depend on Diesel, which would be a bummer.

Reply:
Yeah, I definitely don't want the UUID registry to depend on Diesel. I actually filed https://github.com/diesel-rs/diesel/discussions/3919, and I still need to get back to that.